src/Core/Search/ControllerAction.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Search;
+
+/**
+ * Utility class to extract information from modern controller FQCN.
+ */
+final class ControllerAction
+{
+    /**
+     * Retrieve the Controller's action and name from a FQCN.
+     *
+     * @param string $controller
+     * @return array
+     */
+    public static function fromString($controller)
+    {
+        return [
+            self::getControllerName($controller),
+            self::getActionName($controller),
+        ];
+    }
+
+    /**
+     * Get current controller name
+     * @param string $controller the full controller name.
+     * @return string
+     */
+    public static function getControllerName($controller)
+    {
+        // Set an offset to avoid matching the first occurrence of Controller.
+        preg_match('~(\w+)Controller$~', $controller, $matches);
```

Comment:
As I said in my previous comments, if the full controller name is like:
`My\NameSpace\MyClass::MyAction`, it will not working and it's better to use `~(\w+)Controller::(?:\w+)Action$~' Otherwise you will never catch anaything. (Or you can explode(::) but it's weird)